Code/Engine/Texture/Image/ImageFormat.h

```diff
@@ -1,312 +1,322 @@
-
-  /// \brief If applicable, returns a bitmask for the alpha component of the format.
-  static ezUInt32 GetAlphaMask(Enum format);
-
-  /// \brief Block width of a compressed format. Defaults to 1 for uncompressed formats.
-  static ezUInt32 GetBlockWidth(Enum format);
-
-  /// \brief Block height of a compressed format. Defaults to 1 for uncompressed formats.
-  static ezUInt32 GetBlockHeight(Enum format);
-
-  /// \brief Block depth of a compressed format. Defaults to 1 for uncompressed formats.
-  static ezUInt32 GetBlockDepth(Enum format);
-
-  /// \brief Returns the data type represented by a format.
-  static ezImageFormatDataType::Enum GetDataType(Enum format);
-
-  /// \brief Returns true if the format is compressed.
-  static bool IsCompressed(Enum format);
-
-  /// \brief Returns true if the format is a depth format.
-  static bool IsDepth(Enum format);
-
-  /// \brief Returns whether the given format is an sRGB format.
-  static bool IsSrgb(Enum format);
-
-  /// \brief Returns true if the format is a stencil format.
-  static bool IsStencil(Enum format);
-
-  /// \brief Returns the corresponding sRGB format if one exists; otherwise returns the unmodified format.
-  static Enum AsSrgb(Enum format);
-
-  /// \brief Returns the corresponding linear format if one exists; otherwise returns the unmodified format.
-  static Enum AsLinear(Enum format);
-
-  /// \brief Computes the number of blocks in X direction (compressed) or pixels (if uncompressed) for a given width (in pixels).
-  static ezUInt32 GetNumBlocksX(Enum format, ezUInt32 width);
-
-  /// \brief Computes the number of blocks in Y direction (compressed) or pixels (if uncompressed) for a given height (in pixels).
-  static ezUInt32 GetNumBlocksY(Enum format, ezUInt32 height);
-
-  /// \brief Computes the number of blocks in Z direction (compressed) or pixels (if uncompressed) for a given height (in pixels).
-  static ezUInt32 GetNumBlocksZ(Enum format, ezUInt32 depth);
-
-  /// \brief Computes the size in bytes of a row of blocks (compressed) or pixels (if uncompressed) of the given width.
-  static ezUInt64 GetRowPitch(Enum format, ezUInt32 width);
-
-  /// \brief Computes the size in bytes of a 2D slice of blocks (compressed) or pixels (if uncompressed) of the given width and height.
-  static ezUInt64 GetDepthPitch(Enum format, ezUInt32 width, ezUInt32 height);
-
-  /// \brief Returns the type of the image format.
-  static ezImageFormatType::Enum GetType(Enum format);
-
-  /// \brief Finds a format matching the given component masks.
-  static ezImageFormat::Enum FromPixelMask(
+#pragma once
+
+#include <Foundation/Types/Types.h>
+#include <Texture/Image/ImageEnums.h>
+#include <Texture/TextureDLL.h>
+
+/// \brief Enum describing the type of an image format.
+struct EZ_TEXTURE_DLL ezImageFormatType
+{
+  /// \brief Enum describing the type of an image format.
+  enum Enum
+  {
+    UNKNOWN,
+    LINEAR,
+    BLOCK_COMPRESSED,
+    PLANAR

```

Comment:
What is a planar format ?

Reply:
YUV with different resolution per plane. We have to discuss this change with Christian next week.